src/Build/BuildCheck/Infrastructure/BuildCheckManagerProvider.cs

```diff
@@ -44,6 +46,8 @@ internal static IBuildComponent CreateComponent(BuildComponentType type)
         return new BuildCheckManagerProvider();
     }
 
+    internal static Stopwatch _stopwatch = new Stopwatch();
```

Comment:
StopWatch is cheap - let's create local variable in each function as needed, to avoid possible issues with overlapped calls